### PR TITLE
Added a new Boyer-Moore implementation in C

### PR DESCRIPTION
**Pull request by thatguy227**
**Changes:**
Added a pointer-based implementation of the Boyer-Moore algorithm using the C language.

### CODE REVIEW

**Comment by thatguy227** [comment]
I will do that after work today. Anything in particular that stands out as non-compliant?


-Gerry


________________________________

File: code/string_algorithms/src/boyer_moore_algorithm/boyer_moore_algorithm2.c [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+{
+	const char * testText;
+	const char * testPattern;
+	const char * result;
+	int spaces;
+	char * matchMarker = "^";
+
+
+	testText = "A string searching example consisting of this sentence.";
+	testPattern = "sting";
```
Comment (abdouskamel) [inline code/string_algorithms/src/boyer_moore_algorithm/boyer_moore_algorithm2.c]: Should be "string" no ?

**Comment by thatguy227** [comment]
No, it should be "sting" so it is found in the text "..consi>>sting<<...


That example is borrowed from the Sedgewick Algorithms textbook.


-Gerry Jacobus


________________________________

File: code/string_algorithms/src/boyer_moore_algorithm/boyer_moore_algorithm2.c (hunk 1 above)
Comment (thatguy227) [reply code/string_algorithms/src/boyer_moore_algorithm/boyer_moore_algorithm2.c]: Needs to be "sting" so it is found in the testText array. The pattern "sting" occurs in the word "consisting" from the text.

Comment (abdouskamel) [reply code/string_algorithms/src/boyer_moore_algorithm/boyer_moore_algorithm2.c]: Right, that's a good example. 👍

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**